### PR TITLE
Add integration sdk core package

### PR DESCRIPTION
**Pull request by charlieduong94**
This adds the `@jupiterone/integration-sdk-core` package. This package just contains some shared types and utilities that an integration would directly use. My hope is to keep this package pretty lean.

Previous PR is #201, next PR is #203

### CODE REVIEW

File: packages/integration-sdk-core/package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "@jupiterone/integration-sdk-core",
+  "version": "0.1.0",
```
Comment (aiwilliams) [inline packages/integration-sdk-core/package.json:3]: At some point along the path, these versions will all match?

File: packages/integration-sdk-core/package.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  "description": "The SDK for developing JupiterOne integrations",
+  "main": "dist/src/index.js",
+  "types": "dist/src/index.d.ts",
+  "repository": "git@github.com:JupiterOne/integration-sdk.git",
```
Comment (aiwilliams) [inline packages/integration-sdk-core/package.json:7]: This is stale now, may be worth addressing as these files are created.

Comment (charlieduong94) [reply packages/integration-sdk-core/package.json]: Good catch, changing now

File: packages/integration-sdk-core/package.json (hunk 1 above)
Comment (charlieduong94) [reply packages/integration-sdk-core/package.json:3]: Yeah, after all of the monorepo changes have been merged, we'll version all of the packages and then publish.

**Pull request merged**